content/pt-br/debugging.md

```diff
@@ -0,0 +1,29 @@
+---
+title: Debugging
+status: Completed
+category: conceito
+tags: ["aplicação", "", ""]
+---
+
+## O que é
+
+Debugging é o processo ou atividade de encontrar e resolver bugs (ou erros) de programas de computador, software ou sistemas para obter o resultado desejado. 
+Um bug é um defeito ou um problema que leva a resultados incorretos ou inesperados.
+
+## Problema relacionado
+
+O desenvolvimento de software é uma atividade complexa que torna quase impossível escrever um código sem introduzir bugs. 
+Esses bugs levam a um código que provavelmente não funcionará como desejado (comportamento indefinido) quando executado. 
+Dependendo de quão crítico é um aplicativo, os bugs podem ter um significativo impacto negativo - financeiramente ou até mesmo em vidas humanas. 
+Normalmente, o código do aplicativo tem que passar por diferentes estágios ou ambientes onde é testado. 
+Quanto mais crítico for um aplicativo, mais preciso o teste deve ser.
+
+## Como isso ajuda
+
+Quando os bugs aparecem, os engenheiros precisam executar o debugging (por exemplo, encontrar e corrigir) no aplicativo para diminuir o comportamento indesejado dos sistemas de produção. 
+O debugging não é uma tarefa fácil, pois os engenheiros precisam rastrear a fonte do comportamento indesejado. 
+Requer conhecimento sobre o próprio código e o contexto de execução em tempo de execução. 
+É aqui que diferentes técnicas e ferramentas de debug são úteis. 
+A análise de logs, traces e métricas, por exemplo, é usada no debugging diretamente na produção. 
```

Comment:
```suggestion
A análise de logs, traces e métricas, por exemplo, é usada no debugging diretamente em ambiente de produção. 
```
I think it may make the text clearer if we add "in production environment" - I feel like in english is ok to keep just 'production', but in portuguese sounds that something is missing.. what do you think?